hummingbot/connector/exchange/chainflip_lp/chainflip_lp_data_formatter.py

```diff
@@ -0,0 +1,282 @@
+import math
+from typing import Dict, List, Literal, Any
+from hummingbot.connector.utils import combine_to_hb_trading_pair
+from hummingbot.connector.exchange.chainflip_lp import chainflip_lp_constants as CONSTANTS 
+
+
+class DataFormatter:
+    @classmethod
+    def hex_str_to_int(cls, data: str):
+        return int(data, 16)
+    @classmethod
+    def format_hex_balance(cls, balance:str, asset:Dict[str, str]):
+        int_balance = cls.hex_str_to_int(balance)
+        precision = cls.format_asset_precision(asset)
+        return int_balance/precision
+    @classmethod
+    def format_amount(cls, amount: float| int, asset:Dict[str, str]):
+        precision = cls.format_asset_precision(asset)
+        long_amount = amount * precision
+        return cls.int_to_hex_str(long_amount)
+    @classmethod
+    def format_price(
+        cls, 
+        price:int, 
+        base_asset:Dict[str, str], 
+        quote_asset:Dict[str, str], 
+        sqrt_price = True
+        ):
+        """
+        for example: to get the price of the ETH/USDC pair from the sqrt_price
+        where the provided price is 4512835869581138250956800
+        We Calculate:
+        current_price = 4512835869581138250956800 / 2**96 
+        current_price = current_price ** 2 (provided the price is a square-root price)
+        formated_price = (current_price * base_asset_precision)/quote_asset_precision
+        """
+        base_precision = cls.format_asset_precision(base_asset)
+        quote_precision = cls.format_asset_precision(quote_asset)
+        current_price = price /(2 ** CONSTANTS.FRACTIONAL_BITS)
+        if sqrt_price:
+            current_price = current_price ** 2
+        formated_price = (current_price * base_precision)/quote_precision
+        return formated_price
+    
+    @classmethod
+    def int_to_hex_str(cls, data: int):
+        return str(hex(data))
+    @classmethod
+    def format_order_response(cls, response, base_asset, quote_asset):
+        data = response["result"]
+        limit_orders = data["limit_orders"]
+        asks = []
+        bids = []
+        for order in limit_orders["asks"]:
+            price = cls.format_hex_balance(
```

Comment:
the variable name seems wrong here. the `original_sell_amount` is the amount of the asset that is sold in this order

Reply:
Both asks and bids responses uses `original_sell_amount` so that is why  i used is there. i did not see `original_buy_amount`. also, is it the `original_sell_amount` or the `sell_amount` that is the amount of the order that was placed by the trader?

Reply:
the variable is called `price`, so neither the `original_sell_amount` or the `sell_amount` seem to be appropriate. if you want to calculate the price of an order the `tick` contains the information

Reply:
Could you help me with an algorithm for this?

Reply:
i think you already started a `convert_tick_to_price` method for this? the logic should look similar to this
```
rawPrice = 1.0001 ** tick
humanReadablePrice = rawPrice * (10**ethDecimals) / (10**usdcDecimals)
```